ADR/0026-specifying-ocp-targets-for-fbc.md

```diff
@@ -0,0 +1,72 @@
+# 26. Specifying OCP targets for File-based Catalogs
+
+* Date 2023-06-08
+
+## Status
+
+Accepted
+
+## Context
+
+One of the supported component types within AppStudio are[File-based Catalogs (FBC)].
+These catalogs can either be used in isolation with a version of `opm` packaged in the container
+itself or in conjunction with other catalog configurations via a service such as
+[IIB Image Builder]. Red Hat OpenShift Container Platform (OCP) is one example of a platform that
+leverages FBCs for defining the operator graphs. In order to enable operator support to vary on a
+version-by-version basis, Red Hat maintains one catalog per OpenShift version.
+
+In order to support being able to target FBC components to specific versions of Red Hat OpenShift,
+AppStudio needs to be able to keep track of the specific targeted version. In addition to the concerns
+around releasing FBC components to OpenShift, the version of `opm` used by each version of OpenShift
+may differ, so the AppStudio integration process will need to ensure that tests are run using an appropriate
+binary version.
+
+## Decision
+
+All FBC components intending to be released to OCP will be built using a OCP-specific parent image containing
+the target version number as a tag. This will result in a `FROM` instruction like
+
+```
+# The base image is expected to contain
+# /bin/opm (with a serve subcommand) and /bin/grpc_health_probe
+FROM registry.redhat.io/openshift4/ose-operator-registry:v4.12
+```
+
+Any image build process for the FBC components will need to include an "org.opencontainers.image.base.name"
+annotation indicating the pullspec of the base image. This will enable all components to use `skopeo` to inspect
+the artifact to retrieve the pullspec
+
+```bash
+$ skopeo inspect --raw docker://quay.io/redhat-user-workloads/<...> |
+jq
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "config": {
+    "mediaType": "application/vnd.oci.image.config.v1+json",
+  [...]
+  "annotations": {
+    "org.opencontainers.image.base.digest": "[...]",
+    "org.opencontainers.image.base.name": "[...]"
+  }
+}
+```
+
+The target Red Hat OpenShift version will then be able to be pulled from the image tag on the
+"org.opencontainers.image.base.name" annotation. If a task within the AppStudio pipeline needs to access
```

Comment:
can we maybe show the corresponding  value of the annotation given the example FROM line above for completeness?

Reply:
That was my intention. I was trying to build the image from RHTAP as a confirmation for the format that would be present but I was unable to get the registry service account configured in my workspace. I will make this change once I have an example.

Reply:
sounds good. thanks